applications/EigenSolversApplication/custom_solvers/eigen_direct_solver.h

```diff
@@ -166,12 +114,12 @@ class EigenDirectSolver
      */
     void PerformSolutionStep(SparseMatrixType& rA, VectorType& rX, VectorType& rB) override
     {
-        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1> > x(rX.data().begin(), rX.size());
-        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1> > b(rB.data().begin(), rB.size());
+        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1>> x(rX.data().begin(), rX.size());
+        Eigen::Map<Eigen::Matrix<DataType, Eigen::Dynamic, 1>> b(rB.data().begin(), rB.size());
 
-        x = m_solver.solve(b);
+        const bool success = m_solver.Solve(b, x);
 
-        KRATOS_ERROR_IF(m_solver.info() != Eigen::Success) << "Solving failed!" << std::endl;
+        KRATOS_ERROR_IF(!success) << "Decomposition failed!" << std::endl;
```

Comment:
This is only Solution right? decomposition happened in InitializeSolutionStep with m_solver.Compute(a)

Reply:
👍 copy&paste mistake...